### DISCUSSION TITLE
getting confused in retrieving

### DISCUSSION THREAD

**Discussion by Satya9-9** [Cyfrin/foundry-full-course-cu#2125] [General]
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.25;

/*Create a smart contract that can store and view a list of animals. Add manually
three (3) animals and give the possibility to the user to manually add an 
indefinite number of animals into the smart contract.*/
contract Animal{
    uint256[3] listofanimals;

    struct Animals{
        string animal;
    }

    Animals public goat = Animals("goat");
    Animals public ox = Animals("ox");
    Animals public cat = Animals("cat");

    uint256 animalcount = 0;
    
    function addAnimal(string memory _Animal) public {
        listofanimals[animalcount]=_Animal;
        animalcount++;
     }

    function retrieve () public view returns() {
        return listofanimals;
    }

} what should i write inside the brackets of returns cause its throwing an error

**Comment by EngrPips** [comment]
Hello @ Satya-1729, We have resolved this on Discord already, right?

**Reply by Satya9-9** [reply]
yes thank you

**Reply by EngrPips** [reply]
You are welcome, Friend.
